[test] Include H2-1E checks in the tests

ferbetanzo opened the pull request:
This PR modifies the existing tests to add checks for the H2-1E machine. The changes are:

### conftest.py
- New `pytest_make_parametrize_id` to customize the test name for parametrized tests.

### backend_test.py
- `test_cost_estimate`: fix so that the different backends are tested and the correct error messages asserted.
- `test_simulator`: use `H2-1E` instead of `H1-2E`.
- `test_batching`: use `H2-1E` instead of `H1-2E`.

cqc-alec commented on docs/changelog.rst:
```diff
@@ -15,6 +15,7 @@ Unreleased
   feature.)
 * Use "q" instead of "node" as the name of the single qubit register in compiled
   circuits.
+* Add checks for H2-1E in ``tests/backend_test.py``.
```
I don't think it needs a changelog entry as there's nothing user-facing here.

cqc-alec replied:
Hmm, I still see the entry, is github playing tricks?

cqc-alec commented on tests/backend_test.py:
```diff
@@ -505,9 +531,10 @@ def test_shots_bits_edgecases(n_shots, n_bits) -> None:
     assert res.get_counts() == correct_counts
 
 
+@pytest.mark.skip("Will be moved to a set of long running integration tests")
```
Is there already an issue to capture this task?

ferbetanzo replied:
There is one now :) #147

cqc-alec reviewed (commented):
Still seems to be in changelog?

ferbetanzo commented:
> Still seems to be in changelog?

Sorry, it seems there was an issue while doing the merge with develop. I've removed the entry and fixed the issues :)
